### PR TITLE
 OCPCLOUD-2565: Featuregate implementation

### PR DESCRIPTION
**Pull request by theobarberbany**
This change: 

- Updates the MAO, to bring in new changes to the `Machine` Reconciler.
- Reads in feature gates from flags, passing them to the reconciler. 

Running the binary without passing the --feature-gates flag will result in the feature gates initialising as follows: 

```
I0812 17:36:32.555187    1120 main.go:154] Initializing feature gates: AllAlpha=true|false (ALPHA - default=false), AllBeta=true|false (BETA - default=false), MachineAPIMigration=true|false (ALPHA - default=false)
I0812 17:36:32.555275    1120 main.go:159] FeatureGateMachineAPIMigration initialised: false
```

Note that any feature gates that are GA will not be printed in the help text, but can still be set by flags.

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -135,7 +146,16 @@ func main() {
...
 	}
 
+	// Sets feature gates from flags
+	klog.Infof("Initializing feature gates: %s", strings.Join(defaultMutableGate.KnownFeatures(), ", "))
+	err = defaultMutableGate.SetFromMap(featureGateArgs)
+	if err != nil {
+		klog.Fatalf("Error setting feature gates from flags: %v", err)
+	}
+	klog.Infof("FeatureGateMachineAPIMigration initialised: %t", defaultMutableGate.Enabled(featuregate.Feature(apifeatures.FeatureGateMachineAPIMigration)))
+	klog.Infof("FeatureGateAzureWorkloadIdentity initialised: %t", defaultMutableGate.Enabled(featuregate.Feature(apifeatures.FeatureGateAzureWorkloadIdentity)))
```
Comment (theobarberbany) [inline cmd/manager/main.go:156]: I think we may want to remove the existing feature gating here in favour of passing by flags, as that's what we've done for vsphere

**Comment by nrb** [comment]
I notice that https://github.com/openshift/machine-api-provider-aws/pull/109 updates an associated test file, but this PR does not. Do we need that test in every updated repository?

**Comment by theobarberbany** [comment]
@nrb  I don't think so. IIUC it's because MAPA contains a test that creates a `Machine` controller, and all of the other providers do not. 

I think we get this coverage from the machine controller tests that run on the MAO repo.

**Pull request merged**